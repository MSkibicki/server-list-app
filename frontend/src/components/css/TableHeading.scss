@import "./Global.scss";

.table-heading {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: 59.8%;
  background-color: $white;
  border: 2px solid $list-border-color;

  .name {
    width: 40%;
    color: $table-heading-color;
    font-weight: 600;
    text-transform: uppercase;
    font-size: $nav-font;
    margin-left: 3em;
  }

  .status {
    color: $table-heading-color;
    font-weight: 600;
    text-transform: uppercase;
    font-size: $nav-font;
  }
}

@media (max-width: 500px) {
  .table-heading {
    width: 79.6%;
  }
}
