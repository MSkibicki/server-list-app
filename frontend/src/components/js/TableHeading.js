import React from "react";
import "../css/TableHeading.scss";

const TableHeading = () => {
  return (
    <div className="table-heading">
      <p className="name">name</p>
      <p className="status">status</p>
    </div>
  );
};

export default TableHeading;
