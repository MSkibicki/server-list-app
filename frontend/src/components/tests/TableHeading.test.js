import TableHeading from "./../js/TableHeading";

describe("<TableHeading />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<TableHeading />);

    expect(wrapper).toMatchSnapshot();
  });

  it("has correct div classname", () => {
    const wrapper = shallow(<TableHeading />);

    const divClass = wrapper.find("div").hasClass("table-heading");

    expect(divClass).toBe(true);
  });

  it("has correct name paragraph classname", () => {
    const wrapper = shallow(<TableHeading />);

    const paragraphNameClass = wrapper.find("p.name").hasClass("name");

    expect(paragraphNameClass).toBe(true);
  });

  it("has correct status paragraph classname", () => {
    const wrapper = shallow(<TableHeading />);

    const paragraphStatusClass = wrapper.find("p.status").hasClass("status");

    expect(paragraphStatusClass).toBe(true);
  });

  it("displays paragraph text correctly", () => {
    const wrapper = shallow(<TableHeading />);

    const text = wrapper.find("p.name").text();

    expect(text).toEqual("name");
  });

  it("displays paragraph text correctly", () => {
    const wrapper = shallow(<TableHeading />);

    const text = wrapper.find("p.status").text();

    expect(text).toEqual("status");
  });
});
