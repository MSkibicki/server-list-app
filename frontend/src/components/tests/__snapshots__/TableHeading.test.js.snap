// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableHeading /> renders correctly 1`] = `
<div
  className="table-heading"
>
  <p
    className="name"
  >
    name
  </p>
  <p
    className="status"
  >
    status
  </p>
</div>
`;
